sdk/eventhub/eventhubs-checkpointstore-blob/src/blobCheckpointStore.ts

```diff
@@ -0,0 +1,254 @@
+   *
+   * @param partitionOwnership The list of partition ownership this instance is claiming to own.
+   * @return A list partitions this instance successfully claimed ownership.
+   */
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    let partitionOwnershipArray: PartitionOwnership[] = [];
+    for (const ownership of partitionOwnership) {
+      const blobName = BlobCheckpointStore.getBlobPrefix({ type: "ownership", ...ownership });
+      try {
+        let updatedBlobResponse = await this._setBlobMetadata(
+          blobName,
+          {
+            ownerid: ownership.ownerId
+          },
+          ownership.etag
+        );
+
+        if (updatedBlobResponse.lastModified) {
+          ownership.lastModifiedTimeInMs = updatedBlobResponse.lastModified.getTime();
+        }
+
+        ownership.etag = updatedBlobResponse.etag;
+        partitionOwnershipArray.push(ownership);
+        log.blobCheckpointStore(
+          `[${ownership.ownerId}] Claimed ownership successfully for partition: ${ownership.partitionId}`,
+          `LastModifiedTime: ${ownership.lastModifiedTimeInMs}, ETag: ${ownership.etag}`
+        );
+      } catch (err) {
+        // NOTE: there is some ordinary contention that can occur as different consumers battle over
+        // ownership. So the catching (and _only_ logging, not rethrowing) of this error is intentional.
+        log.error(
+          `Error occurred while claiming ownership for partition: ${ownership.partitionId}`,
+          err
+        );
+      }
+    }
+    return partitionOwnershipArray;
+  }
+
+  /**
+   * Lists all the checkpoints in a data store for a given namespace, eventhub and consumer group.
+   *
+   * @param fullyQualifiedNamespace The fully qualified Event Hubs namespace. This is likely to be similar to
+   * <yournamespace>.servicebus.windows.net.
+   * @param eventHubName The event hub name.
+   * @param consumerGroup The consumer group name.
+   */
+  async listCheckpoints(
+    fullyQualifiedNamespace: string,
+    eventHubName: string,
+    consumerGroup: string
+  ): Promise<Checkpoint[]> {
+    const blobPrefix = BlobCheckpointStore.getBlobPrefix({
+      type: "checkpoint",
+      fullyQualifiedNamespace,
+      eventHubName,
+      consumerGroup
+    });
+
+    const blobs = this._containerClient.listBlobsFlat({
+      includeMetadata: true,
+      prefix: blobPrefix
+    });
+
+    const checkpoints: Checkpoint[] = [];
+
+    for await (const blob of blobs) {
+      const blobPath = blob.name.split("/");
+      const blobName = blobPath[blobPath.length - 1];
+
+      const checkpointMetadata = blob.metadata as CheckpointMetadata;
+
+      checkpoints.push({
+        consumerGroup,
+        eventHubName,
+        fullyQualifiedNamespace,
+        partitionId: blobName,
+        offset: parseInt(checkpointMetadata.offset, 10),
+        sequenceNumber: parseInt(checkpointMetadata.sequencenumber, 10)
```

Comment:
One thing to note is that these values could be `NaN` if they are undefined, empty string or a string that's not a number. We should make sure our EventProcessor code can handle situations where `NaN` is the value.

Reply:
Added in more checks and throwing more aggressively since incorrectly formatted blob metadata indicates a rather serious issue.